apps/lib/apps.c

```diff
@@ -2681,6 +2681,57 @@ int set_cert_times(X509 *x, const char *startdate, const char *enddate,
     return 1;
 }
 
+int set_crl_lastupdate(X509_CRL *crl, const char *lastupdate)
+{
+    int ret = 0;
+    ASN1_TIME *tm = ASN1_TIME_new();
+
+    if (tm == NULL)
+        goto end;
+
+    if (lastupdate == NULL) {
+        if (X509_gmtime_adj(tm, 0) == NULL)
+            goto end;
+    } else {
+        if (!ASN1_TIME_set_string_X509(tm, lastupdate))
+            goto end;
+    }
+
+    if (!X509_CRL_set1_lastUpdate(crl, tm))
+        goto end;
+
+    ret = 1;
+end:
+    ASN1_TIME_free(tm);
+    return ret;
+}
+
+int set_crl_nextupdate(X509_CRL *crl, const char *nextupdate,
+                       int days, int hours, int secs)
```

Comment:
days, hours, secs were long params before.. Should they still be?

Reply:
Whoops, my mistake - fixed in the latest revision, thanks.